sklearn/linear_model/glm.py

```diff
@@ -0,0 +1,1813 @@
+        return self._include_lower_bound
+
+    @property
+    def include_upper_bound(self):
+        return self._include_upper_bound
+
+    def unit_variance(self, mu):
+        return 1 + mu**2
+
+    def unit_variance_derivative(self, mu):
+        return 2*mu
+
+    def unit_deviance(self, y, mu):
+        return (2*y*(np.arctan(y) - np.arctan(mu)) +
+                np.log((1+mu**2)/(1+y**2)))
+
+
+def _irls_step(X, W, P2, z):
+    """One step in iteratively reweighted least squares
+
+    Solve A w = b for w with
+    A = (X' W X + P2)
+    b = X' W z
+    z = eta + D^-1 (y-mu)
+
+    See also fit method of :class:`GeneralizedLinearRegressor`.
+
+    Parameters
+    ----------
+    X : {numpy array, sparse matrix}, shape (n_samples, n_features)
+        Training data (with intercept included if present)
+
+    W : numpy array, shape (n_samples,)
+
+    P2 : {numpy array, sparse matrix}, shape (n_features, n_features)
+        The L2-penalty matrix or vector (=diagonal matrix)
+
+    z  : numpy array, shape (n_samples,)
+        Working observations
+
+    Returns
+    -------
+    coef: array, shape (X.shape[1])
+    """
+    # TODO: scipy.linalg.solve seems faster, but ordinary least squares uses
+    #       scipy.linalg.lstsq. What is more appropriate?
+    n_samples, n_features = X.shape
+    if sparse.issparse(X):
+        W = sparse.dia_matrix((W, 0), shape=(n_samples, n_samples)).tocsr()
+        if P2.ndim == 1:
+            L2 = (sparse.dia_matrix((P2, 0), shape=(n_features, n_features))
+                  ).tocsr()
+        else:
+            L2 = sparse.csr_matrix(P2)
+        XtW = X.transpose() * W
+        A = XtW * X + L2
+        b = XtW * z
+        coef = sparse.linalg.spsolve(A, b)
+    else:
+        XtW = (X.T * W)
+        A = XtW.dot(X)
+        if P2.ndim == 1:
+            A[np.diag_indices_from(A)] += P2
+        else:
+            A += P2
+        b = XtW.dot(z)
+        coef = linalg.solve(A, b)
+    return coef
+
+
+class GeneralizedLinearRegressor(BaseEstimator, RegressorMixin):
+    """Regression via a Generalized Linear Model (GLM) based on reproductive
+    Exponential Dispersion Models (EDM) with combined L1 and L2 priors as
+    regularizer.
+
+    Minimizes the objective function::
+
+            1/(2s) * deviance(y, h(X*w))
+            + alpha * l1_ratio * ||P1*w||_1
+            + 1/2 * alpha * (1 - l1_ratio) * w*P2*w
+
+    with inverse link function `h` and s=sum of `sample_weight` (which equals
+    n_samples for `sample_weight=None`).
+    For `P1=P2=identity`, the penalty is the elastic net::
+
+            alpha * l1_ratio * ||w||_1
+            + 1/2 * alpha * (1 - l1_ratio) * ||w||_2^2
+
+    If you are interested in controlling the L1 and L2 penalty
+    separately, keep in mind that this is equivalent to::
+
+            a * L1 + b * L2
+
+    where::
+
+            alpha = a + b and l1_ratio = a / (a + b)
+
+    The parameter `l1_ratio` corresponds to alpha in the glmnet R package while
```

Comment:
belongs in the user guide, or in parameter documetnation

Reply:
I followed the example of
https://github.com/scikit-learn/scikit-learn/blob/db17f3e2221fb6cec256d2d3501e259c5d5db934/sklearn/linear_model/coordinate_descent.py#L509-L512
Furthermore, this is also in the user guide. Would you prefer to have it removed here? I'd like to keep at least the objective function.